provider/core/src/request.rs

```diff
@@ -458,48 +464,12 @@ impl DataLocale {
 }
 
 impl DataLocale {
-    /// Returns whether this [`DataLocale`] has all empty fields (no components).
-    ///
-    /// See also:
-    ///
-    /// - [`DataLocale::is_und()`]
-    /// - [`DataLocale::is_langid_und()`]
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use icu_provider::DataLocale;
-    ///
-    /// assert!("und".parse::<DataLocale>().unwrap().is_empty());
-    /// assert!(!"und-u-ca-buddhist"
-    ///     .parse::<DataLocale>()
-    ///     .unwrap()
-    ///     .is_empty());
-    /// assert!(!"ca-ES".parse::<DataLocale>().unwrap().is_empty());
-    /// ```
-    pub fn is_empty(&self) -> bool {
-        self == <&DataLocale>::default()
-    }
```

Comment:
Nit: I think I prefer `is_empty` since that is what we use in Language and LanguageIdentifier. Also, `is_und` is ambiguous because does it return true or false for something like `und-CH`?

Reply:
It's called `is_und` on `LanguageIdentifier.` The language identifier is `"und"`, it's not empty (`""`).

Reply:
https://unicode-org.github.io/icu4x/rustdoc/icu_locale_core/struct.LanguageIdentifier.html#method.is_empty

I won't block the PR on this.

Reply:
Oh whoops. This is unreleased, and I think `is_und` is a better name, so I'd rather change `LanguageIdentifier`.

Reply:
This PR introduces an inconsistency which should be resolved before 2.0. I approved the LanguageIdentifier change a month or so ago because I think `is_empty` is the correct name. It sounds like we need to bikeshed.